framework/test_selectors/SelectorFactory.cpp

```diff
@@ -206,6 +207,26 @@ void generate_test_list(std::vector<struct test *> &test_list, std::span<struct
     }
 }
 
+std::optional<const std::span<struct test * const>> auto_detect_test_list(const std::vector<const TestList*>& test_lists,
+                                                                          const TestList* default_test_list)
+{
+#ifdef SANDSTONE
```

Comment:
If I do not use this flag correctly, then it will be better to create separate file for this specific function auto_detect_test_list(), so that it won't be included in the unittests source files

Reply:
That's ok, the same happens above (line 83)

I actually prefer to have it in this file, instead of generated,  because we do need to apply the sorting. The current generated code for the internal product has:
```c++
    &icx_test_list,
    &cpx_test_list,
    &skx_test_list,
    &spr_test_list,
```

This is not in any special sorting order (ICX later than CPX and SKX, but earlier than SPR). So we must either sort in this function (at a minimum to `assert` that the supplied order is correct).

Reply:
But I thought you agreed that there can be no sorting at all, as no method ensures 100% correct sorting. User should define the architecture lists in the correct order.

Reply:
We can go with that. If all lists are supersets of previous ones, we could search for the one that matches most bits... but the ISA extensions weren't a monotonic sequence. So I agree on enforcing the order on whoever supplies the lists.